File: nysol_python/mcmd/methods/convTex/db/smp/mtra.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mtra'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
customer,item
A,a
A,b
B,c
B,d
B,e
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
``customer`` を単位に ``item`` をスペース区切りで結合し、
``transaction`` という項目名で出力する。
'''
script['sh_code']='''
mtra k=customer f=item:transaction i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mtra(k="customer", f="item:transaction", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='アイテムの区切り文字を-(ハイフン）で実行'
script['comment']='''
'''
script['sh_code']='''
mtra k=customer f=item:transaction delim=- i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mtra(k="customer", f="item:transaction", delim="-", i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='アイテムを降順に並べ替えてから変換'
script['comment']='''
'''
script['sh_code']='''
mtra k=customer s=item%r f=item:transaction i=dat1.csv o=rsl3.csv
'''
script['py_code']='''
nm.mtra(k="customer", s="item%r", f="item:transaction", i="dat1.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mtra.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
customer,item
A,a
A,b
B,c
B,d
B,e
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Combine the corresponding \verb|item| for each \verb|customer| in a string using a space as the delimiter, and save output in the column named \verb|transaction|.
EOF
scp=<<'EOF'
more dat1.csv
mtra k=customer f=item:transaction i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Use hyphen (-) as item delimiter"
comment=<<'EOF'
EOF
scp=<<'EOF'
mtra k=customer f=item:transaction delim=- i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)


############## 例3
title="Convert items in descending sort order"
comment=<<'EOF'
EOF
scp=<<'EOF'
mtra k=customer s=item%r f=item:transaction i=dat1.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)